Tests/PackageLoading/ConventionTests.swift

```diff
@@ -552,6 +641,36 @@ class ConventionTests: XCTestCase {
         }
     }
 
+    func testNoSourcesInModule() throws {
+        let fs = InMemoryFileSystem()
+        try fs.createDirectory(AbsolutePath("/Sources/Module"), recursive: true)
+
+        PackageBuilderTester("MyPackage", in: fs) { result in
+            result.checkDiagnostic("the module at /Sources/Module does not contain any source files fix: either remove the module folder, or add a source file to the module")
```

Comment:
Also OT, but lately I've wondered whether this should be a warning not an error.

I find it really annoying when stashing sources for a new module, but stray editor temp files make the directory stay around and I have to clean it before SwiftPM is willing to build.